source/common/http/conn_manager_impl.cc

```diff
@@ -388,6 +388,10 @@ uint64_t ConnectionManagerImpl::ActiveStream::connectionId() {
   return connection_manager_.read_callbacks_->connection().id();
 }
 
+const Network::Connection* ConnectionManagerImpl::ActiveStream::connection() {
+  return &connection_manager_.read_callbacks_->connection();
+}
+
 Ssl::Connection* ConnectionManagerImpl::ActiveStream::ssl() {
   return connection_manager_.read_callbacks_->connection().ssl();
```

Comment:
Yes good catch. We should definitely kill ssl() also and make sure a const connection returns a const ssl connection. @jrajahalme if you feel like cleaning this up also that would be great. Otherwise please put in a TODO and I can take care of this before we ship 1.4.0.